panic: concurrent write to websocket connection

**环境信息**
请根据实际使用环境修改以下信息
go-cqhttp版本: v0.9.22
运行环境: windows_amd64
连接方式: 正向WS

**bug内容**
运行一段时间后偶尔会出现此错误并结束运行

**复现方法**
在不同的机器上均出现过

panic: concurrent write to websocket connection

goroutine 40465 [running]:
github.com/gorilla/websocket.(*messageWriter).flushFrame(0xc001ae06c0, 0x101, 0x
0, 0x0, 0x0, 0x10000c001b3ed00, 0x0)
        /go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:610 +0x5f0
github.com/gorilla/websocket.(*messageWriter).Close(0xc001ae06c0, 0x40ca53, 0xc0
00680a80)
        /go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:724 +0x69
github.com/gorilla/websocket.(*Conn).beginMessage(0xc0000f2420, 0xc000353dd0, 0x
1, 0xc000680a00, 0xc001b3eea0)
        /go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:473 +0x274
github.com/gorilla/websocket.(*Conn).WriteMessage(0xc0000f2420, 0x1, 0xc001b3eea
0, 0x183, 0x1a0, 0x1a0, 0x0)
        /go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:757 +0x18c
github.com/Mrs4s/go-cqhttp/server.(*websocketServer).onBotPushEvent(0x13231c0, 0
xc0019afcb0)
        /github/workspace/server/websocket.go:322 +0x305
github.com/Mrs4s/go-cqhttp/coolq.(*CQBot).dispatchEventMessage.func1(0xc0002c229
0, 0xc0019afcb0)
        /github/workspace/coolq/bot.go:224 +0x66
created by github.com/Mrs4s/go-cqhttp/coolq.(*CQBot).dispatchEventMessage
        /github/workspace/coolq/bot.go:222 +0x12d

--- a/server/websocket.go
+++ b/server/websocket.go
@@ -319,6 +319,7 @@ func (s *websocketServer) onBotPushEvent(m coolq.MSG) {
 	for i, l := 0, len(s.eventConn); i < l; i++ {
 		conn := s.eventConn[i]
 		log.Debugf("向WS客户端 %v 推送Event: %v", conn.RemoteAddr().String(), m.ToJson())
+		conn.Lock()
 		if err := conn.WriteMessage(websocket.TextMessage, []byte(m.ToJson())); err != nil {
 			_ = conn.Close()
 			next := i + 1
@@ -330,7 +331,9 @@ func (s *websocketServer) onBotPushEvent(m coolq.MSG) {
 			i--
 			l--
 			conn = nil
+			continue
 		}
+		conn.Unlock()
 	}
 }
 

